sktime/annotation/ggs.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+"""
+Greedy Gaussian Segmentation (GGS).
+
+Based on:
+
+    Hallac, D., Nystrup, P. & Boyd, S.
+    Greedy Gaussian segmentation of multivariate time series.
+    Adv Data Anal Classif 13, 727–751 (2019).
+    https://doi.org/10.1007/s11634-018-0335-0
+
+- source code adapted based on: https://github.com/cvxgrp/GGS
+- paper available at: https://stanford.edu/~boyd/papers/pdf/ggs.pdf
+"""
+
+import logging
+import math
+from typing import Dict, List, Tuple
+
+import numpy as np
+import numpy.typing as npt
+from attrs import asdict, define, field
+from sklearn.utils.validation import check_random_state
+
+logger = logging.getLogger(__name__)
+
+
+@define
```

Comment:
I think using attrs is an *excellent* idea!!  I might try it out as well! Nice.

Reply:
Thanks, for a moment I was thinking about alternatives but this worked quite well. The added benefit is that you need to declare **all** the attributes up front which help with state management of the class and forces a cleaner design. Looking forward to seeing your take on this :+1: